File: internal/sast/fixtures/java/ssti_positive.java
// internal/sast/fixtures/java/ssti_positive.java
import javax.servlet.http.*;
import java.io.*;
import org.apache.velocity.app.VelocityEngine;
import org.apache.velocity.VelocityContext;

public class SstiPositive extends HttpServlet {
    public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
        String template = "Hello $name " + req.getParameter("suffix");
        VelocityEngine engine = new VelocityEngine();
        StringWriter out = new StringWriter();
        engine.evaluate(new VelocityContext(), out, "user", template);
        resp.getWriter().write(out.toString());
    }
}

File: internal/sast/fixtures/python/ssti_positive.py
# internal/sast/fixtures/python/ssti_positive.py
from flask import Flask, request, render_template_string

app = Flask(__name__)

@app.route("/hello")
def hello():
    name = request.args.get("name", "world")
    template = "Hello " + name + "!"
    return render_template_string(template)
